Yoast/javascript
# Improve stemming single syllable words

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds a check for stemming Indonesian single syllable words.
* [Yoast SEO Premium] Improves word form recognition of Indonesian single syllable words.

## Relevant technical choices:

* This PR only concerns Indonesian single syllable words with this form **[menge-] + single syllable word + [-kan/-i]** and **[penge-] +single syllable word + [-an]**, with [] being optional. 

* However, as the result of implementing this check, two syllable words whose stem starts with one of the words in singleSyllableWords list and ends with -kan/-an/ are going to be overstemmed. E.g. lokan -> lo ("lo" is in the list)

* Two syllable words whose stem starts with ke- and ends in one of the single syllable words which gets prefix meng- ( **meng-/peng- + ke + single syllable word + [-kan/-an/-i]**  ). Example, **mengebiri**  is stemmed into **bir** (single syllable word) while it actually should be stemmed into **kebiri** (“ebiri” is listed in kBegining list)
Other example  **mengecap** / **pengecap** is currently stemmed into **cap** which is a valid word, but it can also be stemmed into **kecap** ("ecap" is in the kBeginning list).


## Test instructions

This PR can be tested by following these steps:

Check the unit tests:

* Run yarn test and make sure all tests still pass.
* Add more single syllable word forms in the specs and make sure that they have the same stem. E.g. pengeklik, mengeklik, pengeklikan, mengeklikkan, mengekliki, klikan, klikkan -> klik
pengecekan, pengecek, mengecek, mengecekkan, mengeceki, cekkan -> cek. Other example of single syllable words can be found here Yoast/YoastSEO.js-premium-configuration#186 and add the possible affixes, see pattern above.
* Also see the relevant technical choice for an overview of words that are not going to stemmed correctly.

Check in the content-analysis app:

* Open the content analysis app by running yarn start;
* Insert a text of al least 150 words;
* Set the locale to Indonesian (i.e., id_ID);
* Make sure you have different forms of single syllable words you want to test in the text and set the word you want to test as a keyword (e.g., pengetesan, pengetes, mengetes, mengeteskan, mengetesi, teskan, tes, dites, tesnya, teslah, tesku, teskanlah should be all recognised );
* Check on the eye icon that all the word forms become highlighted

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-440
